sky/utils/common_utils.py

```diff
@@ -116,6 +122,38 @@ def _base36_encode(num: int) -> str:
     return _base36_encode(int_value)
 
 
+def adjust_cluster_name(cluster_name: str) -> str:
+    adjusted_cluster_name_arr = []
+    for ch in cluster_name:
+        if ch.isalnum() or ch == '-':
+            adjusted_cluster_name_arr.append(ch.lower())
+    if not adjusted_cluster_name_arr[0].isalpha():
+        adjusted_cluster_name_arr.insert(0, CLUSTER_NAME_VALID_PREFIX)
```

Comment:
How about prepending `sky-` instead?

Reply:
Changed to prepend `sky-`. Note that the default max length of cluster name on the cloud is 15 so `sky-` will take up a nontrivial amount of these characters https://github.com/skypilot-org/skypilot/blob/bf594274c8767356893060302e1b40810b42d86c/sky/utils/common_utils.py#L161